### PR TITLE
Fix update script and add sha256 validation

### LINKED ISSUE #36: Ruby archive downloaded over insecure link
Ruby archives are downloaded over HTTP and checksums are not verified. Checksums are published on official website, and since the versions seem hardcoded in `Dockerfile`, the checksums could also be added.

### PR DESCRIPTION
**Pull request by Moghedrin**
Fixes #36
For real this time - I promise.

### CODE REVIEW

**Comment by yosifkit** [comment]
LGTM

File: 2.0/Dockerfile [hunk 1]
```diff
@@ -9,8 +10,10 @@ RUN apt-get update \
 	&& apt-get install -y bison libgdbm-dev ruby \
 	&& rm -rf /var/lib/apt/lists/* \
 	&& mkdir -p /usr/src/ruby \
-	&& curl -SL "http://cache.ruby-lang.org/pub/ruby/$RUBY_MAJOR/ruby-$RUBY_VERSION.tar.bz2" \
-		| tar -xjC /usr/src/ruby --strip-components=1 \
+	&& curl -SL -o ruby.tar.gz "http://cache.ruby-lang.org/pub/ruby/$RUBY_MAJOR/ruby-$RUBY_VERSION.tar.gz" \
+	&& echo "$RUBY_DOWNLOAD_SHA256 *ruby.tar.gz" | sha256sum -c - \
+	&& tar -xzf ruby.tar.gz -C /usr/src/ruby --strip-components=1 \
+	&& rm ruby.tar.gz
```
Comment (tianon) [inline 2.0/Dockerfile]: :wink:

File: update.sh [hunk 2]
```diff
@@ -8,6 +8,7 @@ if [ ${#versions[@]} -eq 0 ]; then
 	versions=( */ )
 fi
 versions=( "${versions[@]%/}" )
+shaPage=$(curl -s 'https://www.ruby-lang.org/en/downloads/')
```
Comment (tianon) [inline update.sh]: https://github.com/docker-library/official-images/issues/628 is definitely relevant here -- just `curl -s` will fail silently and without even an error message if there's any trouble

File: 2.0/Dockerfile (hunk 1 above)
Comment (yosifkit) [inline 2.0/Dockerfile]: stupid docker... we need the fail on unknown lines.

File: update.sh (hunk 2 above)
Comment (yosifkit) [inline update.sh]: I would think that is more pertinent in Dockerfiles, rather than update scripts.

**Comment by Moghedrin** [comment]
@tianon Good catches.

File: 2.0/Dockerfile (hunk 1 above)
Comment (tianon) [inline 2.0/Dockerfile]: We got it, though! :D  (will be in 1.6)

**Comment by tianon** [comment]
LGTM :+1:

**Comment by yosifkit** [comment]
hooray
Hooray!

**Pull request merged**